### ISSUE TITLE
Trouble with Applying PnP Starter Kit Template

### ISSUE THREAD

**Issue by mwsaunders** [pnp/sp-starter-kit#296]
#### Category
- [x ] Question
- [ ] Bug
- [ ] Enhancement

I have been struggling to get the Starter Kit deployed properly.
I am doing this on a MSFT Developer Tenant.
I finally got the SPPKG file deployed in the tenant and see the available web parts.
(Note: I had deployed one of the web parts as a standalone - removing that helped)
I have updated my SP PnP PowerShell.

I have tried both the Apply-PnpTenantTemplate and the Web SharePoint PNP Provisioning Service.  I have removed the sites after the attempts including using PS to permanently remove the team sites.

This is the current error I am receiving.
PS C:\sp-starter-kit-master\provisioning> Apply-PnPTenantTemplate -Path starterkit.pnp -Parameters @{"SiteUrlPrefix"="demo1"}
Apply-PnPTenantTemplate : {"DocumentsUrl":null,"ErrorMessage":null,"GroupId":"f112fdea-3b56-4760-8e40-5c9491cc3fcc","SiteStatus":1,"SiteUrl":null}
At line:1 char:1
+ Apply-PnPTenantTemplate -Path starterkit.pnp -Parameters @{"SiteUrlPr ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Apply-PnPTenantTemplate], Exception
    + FullyQualifiedErrorId : EXCEPTION,SharePointPnP.PowerShell.Commands.Provisioning.Tenant.ApplyTenantTemplate

After this, I have the three sites created.
But they are all empty of anything but the generic home page.
The team sites are not associated to hub site.

=======================================================
Trying again with debug on I get this.
PS C:\sp-starter-kit-master\provisioning> Set-PnPTraceLog -On -Level Debug
PS C:\sp-starter-kit-master\provisioning> Apply-PnPTenantTemplate -Path starterkit.pnp -Parameters @{"SiteUrlPrefix"="demo1"}
powershell.exe Information: 0 : 2019-06-21 10:31:51.3987        [OfficeDevPnP.Core]     [0]     [Information]   File starterkit.pnp retrieved from folde
r       0ms
powershell.exe Information: 0 : 2019-06-21 10:31:51.5283        [OfficeDevPnP.Core]     [0]     [Information]   File starterkit.pnp retrieved from folde
r       0ms
powershell.exe Information: 0 : 2019-06-21 10:31:52.0081        [OfficeDevPnP.Core]     [0]     [Information]   File starterkit.xml retrieved from folde
r       0ms
powershell.exe Information: 0 : 2019-06-21 10:31:52.1168        [Provisioning]  [10]    [Debug] Code execution scope started    0ms     310a6da8-81b3-403f-93a0-c2686a2e27c9
powershell.exe Information: 0 : 2019-06-21 10:31:52.1178        [Provisioning]  [10]    [Information]   ProgressDelegate registered     0ms     310a6da8-81b3-403f-93a0-c2686a2e27c9
powershell.exe Information: 0 : 2019-06-21 10:31:52.1188        [Provisioning]  [10]    [Information]   MessagesDelegate registered     1ms     310a6da8-81b3-403f-93a0-c2686a2e27c9
powershell.exe Information: 0 : 2019-06-21 10:31:52.2413        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-core.en-us.resx retrieved from folder  0ms
powershell.exe Information: 0 : 2019-06-21 10:31:52.2423        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-core.es-es.resx
 retrieved from folder  0ms
powershell.exe Information: 0 : 2019-06-21 10:31:52.2433        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-core.fr-fr.resx
 retrieved from folder  0ms
powershell.exe Information: 0 : 2019-06-21 10:31:52.2433        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-core.de-de.resx
 retrieved from folder  0ms
powershell.exe Information: 0 : 2019-06-21 10:31:52.2443        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-core.nl-nl.resx
 retrieved from folder  0ms
powershell.exe Information: 0 : 2019-06-21 10:31:52.2443        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-core.nb-no.resx
 retrieved from folder  0ms
powershell.exe Information: 0 : 2019-06-21 10:31:52.2453        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-core.sv-se.resx
 retrieved from folder  0ms
powershell.exe Information: 0 : 2019-06-21 10:31:52.2463        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-core.tr-tr.resx
 retrieved from folder  0ms
powershell.exe Information: 0 : 2019-06-21 10:31:52.2583        [Tenant Settings]       [10]    [Debug] Code execution scope started    0ms     310a6da8
-81b3-403f-93a0-c2686a2e27c9
powershell.exe Information: 0 : 2019-06-21 10:31:53.1285        [OfficeDevPnP.Core]     [0]     [Information]   File sharepoint-starter-kit.sppkg retrieved from folder ..\package      0ms
powershell.exe Information: 0 : 2019-06-21 10:31:57.7001        [OfficeDevPnP.Core]     [0]     [Information]   File collabteamsite.json retrieved from folder resources        0ms
powershell.exe Information: 0 : 2019-06-21 10:31:58.6023        [OfficeDevPnP.Core]     [0]     [Information]   File collabcommunicationsite.json retrieved from folder resources       0ms
powershell.exe Information: 0 : 2019-06-21 10:32:02.0041        [Tenant Settings]       [10]    [Debug] Code execution scope ended      9745ms  310a6da8-81b3-403f-93a0-c2686a2e27c9
powershell.exe Information: 0 : 2019-06-21 10:32:02.0161        [Term Groups]   [10]    [Debug] Code execution scope started    0ms     310a6da8-81b3-403f-93a0-c2686a2e27c9
powershell.exe Information: 0 : 2019-06-21 10:32:03.4902        [Term Groups]   [10]    [Debug] Code execution scope ended      1474ms  310a6da8-81b3-403f-93a0-c2686a2e27c9
powershell.exe Information: 0 : 2019-06-21 10:32:03.5051        [Provisioning]  [10]    [Debug] Code execution scope started    0ms     310a6da8-81b3-403f-93a0-c2686a2e27c9
powershell.exe Information: 0 : 2019-06-21 10:32:09.1568        [Provisioning]  [10]    [Debug] Code execution scope ended      5651ms  310a6da8-81b3-403f-93a0-c2686a2e27c9
powershell.exe Information: 0 : 2019-06-21 10:32:09.1578        [Provisioning]  [10]    [Debug] Code execution scope ended      17040ms 310a6da8-81b3-403f-93a0-c2686a2e27c9
Apply-PnPTenantTemplate : Url of the site is required.
Parameter name: siteUrl
At line:1 char:1
+ Apply-PnPTenantTemplate -Path starterkit.pnp -Parameters @{"SiteUrlPr ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Apply-PnPTenantTemplate], ArgumentException
    + FullyQualifiedErrorId : EXCEPTION,SharePointPnP.PowerShell.Commands.Provisioning.Tenant.ApplyTenantTemplate
==========================================

Any idea what I am missing?

Thank you.

**Comment by RiccardoAmadi** [comment]
I'm receiving the same error:

```
Apply-PnPTenantTemplate : A task was canceled.
At line:1 char:1
+ Apply-PnPTenantTemplate -Path starterkit.pnp -Parameters @{"SiteUrlPr ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Apply-PnPTenantTemplate], TaskCanceledException
    + FullyQualifiedErrorId : EXCEPTION,SharePointPnP.PowerShell.Commands.Provisioning.Tenant.ApplyTenantTemplate
```

But I don't see any site created or the web parts available

**Comment by vishalramiit** [comment]
i'm getting the same error as @RiccardoAmadi. Any suggestions?

**Comment by mbruurs** [comment]
Las have the same issue "A task was canceled"

**Comment by VitoDeCarlo** [comment]
I'm having the same issue, as well.  Should I revert back to an older version?

**Comment by gallantdon** [comment]
I am having the same issue as well. 

PS ..\sp-starter-kit-master\provisioning> Apply-PnPTenantTemplate -Path starterkit.pnp -Parameters @{"SiteUrlPrefix"="demo"}
Apply-PnPTenantTemplate : A task was canceled.
At line:1 char:1
+ Apply-PnPTenantTemplate -Path starterkit.pnp -Parameters @{"SiteUrlPr ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Apply-PnPTenantTemplate], TaskCanceledException
    + FullyQualifiedErrorId : EXCEPTION,SharePointPnP.PowerShell.Commands.Provisioning.Tenant.ApplyTenantTemplate

**Comment by treymcs** [comment]
I'm seeing this same issue on our production tenant. I was able to provision the site successfully on a development tenant, but am stuck on this error when attempting our active one. It does appear a timeout is occurring and the app store contains the package and it is indicated as deployed before it fails. Something blocking the actual creation of the sites perhaps? This is really frustrating.

PS G:\git\repos\sp-starter-kit\provisioning> Apply-PnPTenantTemplate -Path starterkit.pnp
powershell.exe Information: 0 : 2019-12-11 12:40:04.2031        [OfficeDevPnP.Core]     [0]     [Information]   File starterkit.pnp retriev
ed from folder  0ms
powershell.exe Information: 0 : 2019-12-11 12:40:04.2956        [OfficeDevPnP.Core]     [0]     [Information]   File starterkit.pnp retriev
ed from folder  0ms
powershell.exe Information: 0 : 2019-12-11 12:40:04.9159        [OfficeDevPnP.Core]     [0]     [Information]   File starterkit.xml retriev
ed from folder  0ms
powershell.exe Information: 0 : 2019-12-11 12:40:05.1369        [Provisioning]  [14]    [Debug] Code execution scope started    0ms     696
62b2e-b389-44d6-b79b-851f2f3d8e90
powershell.exe Information: 0 : 2019-12-11 12:40:05.1374        [Provisioning]  [14]    [Information]   ProgressDelegate registered     0ms
69662b2e-b389-44d6-b79b-851f2f3d8e90
powershell.exe Information: 0 : 2019-12-11 12:40:05.1380        [Provisioning]  [14]    [Information]   MessagesDelegate registered     1ms
69662b2e-b389-44d6-b79b-851f2f3d8e90
powershell.exe Information: 0 : 2019-12-11 12:40:05.3030        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-co
re.en-us.resx retrieved from folder     0ms
powershell.exe Information: 0 : 2019-12-11 12:40:05.3035        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-co
re.es-es.resx retrieved from folder     0ms
powershell.exe Information: 0 : 2019-12-11 12:40:05.3040        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-co
re.fr-fr.resx retrieved from folder     0ms
powershell.exe Information: 0 : 2019-12-11 12:40:05.3050        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-co
re.de-de.resx retrieved from folder     0ms
powershell.exe Information: 0 : 2019-12-11 12:40:05.3055        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-co
re.nl-nl.resx retrieved from folder     0ms
powershell.exe Information: 0 : 2019-12-11 12:40:05.3060        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-co
re.nb-no.resx retrieved from folder     0ms
powershell.exe Information: 0 : 2019-12-11 12:40:05.3065        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-co
re.sv-se.resx retrieved from folder     0ms
powershell.exe Information: 0 : 2019-12-11 12:40:05.3075        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-co
re.tr-tr.resx retrieved from folder     0ms
powershell.exe Information: 0 : 2019-12-11 12:40:05.3080        [ProvisioningTemplate WebHook Call]     [14]    [Debug] Code execution scop
e started       0ms     69662b2e-b389-44d6-b79b-851f2f3d8e90
powershell.exe Information: 0 : 2019-12-11 12:40:05.3085        [ProvisioningTemplate WebHook Call]     [14]    [Debug] Code execution scop
e ended 0ms     69662b2e-b389-44d6-b79b-851f2f3d8e90
powershell.exe Information: 0 : 2019-12-11 12:40:05.3145        [Tenant Settings]       [14]    [Debug] Code execution scope started    0ms
69662b2e-b389-44d6-b79b-851f2f3d8e90
powershell.exe Information: 0 : 2019-12-11 12:40:05.7292        [OfficeDevPnP.Core]     [0]     [Information]   File sharepoint-starter-kit
.sppkg retrieved from folder ..\package 0ms
powershell.exe Information: 0 : 2019-12-11 12:45:05.1979        [Tenant Settings]       [14]    [Debug] Code execution scope ended      2999
85ms    69662b2e-b389-44d6-b79b-851f2f3d8e90
powershell.exe Information: 0 : 2019-12-11 13:16:08.0433        [ProvisioningTemplate WebHook Call]     [14]    [Debug] Code execution scope
 started        0ms     69662b2e-b389-44d6-b79b-851f2f3d8e90
powershell.exe Information: 0 : 2019-12-11 13:16:08.0438        [ProvisioningTemplate WebHook Call]     [14]    [Debug] Code execution scope
 ended  0ms     69662b2e-b389-44d6-b79b-851f2f3d8e90
powershell.exe Information: 0 : 2019-12-11 13:16:08.0443        [Provisioning]  [14]    [Debug] Code execution scope ended      2163640ms
69662b2e-b389-44d6-b79b-851f2f3d8e90
Apply-PnPTenantTemplate : A task was canceled.
At line:1 char:1
+ Apply-PnPTenantTemplate -Path starterkit.pnp
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Apply-PnPTenantTemplate], TaskCanceledException
    + FullyQualifiedErrorId : EXCEPTION,SharePointPnP.PowerShell.Commands.Provisioning.Tenant.ApplyTenantTemplate

**Comment by eoverfield** [comment]
@treymcs although you too are having issues with deployment, your specific error looks different. The time it is taking for aspects to run is far too long. It looks like it took 5 minutes to attempt to install the sppkg, then another 30 minutes for something else to occur. Although this process does take time, well under 10 minutes is expected for the entire process.

Have you attempted to install the kit again? Did you see similar lags? What happens if you upload the sppgk to your app catalog in the browser? Do you see long delays? I am wondering if you got hit with throttling because other processes where hitting your tenant.

If you continue to have issues, I ask that you open a new issue as likely you are running into a new problem.

**Comment by treymcs** [comment]
I tried about 4 times yesterday with similar results. It does seem to be problem with something taking too long and hitting a timeout. I can upload the sppgk to the app browser manually and it didn't appear to be overly long, certainly around 5 mins or so. However, I was able to have some success by using the suggestion of the older commit of pnp file. This seemed like it was going to succeed, getting past the point which the other one hung, but got an error eventually after creating all of the sites. The sites appear to be ok, except the portal wasn't registered as a hub, so I"m declaring victory and working with these as a start. Need to move on at this point.
